src/Service/Auth/auth.service.ts

```diff
@@ -0,0 +1,120 @@
+import { Request, Response, NextFunction } from "express";
+import { StatusCodes } from "http-status-codes";
+import { UserType } from "../../Models/user.model";
+import AuthRepository from "../../Repository/auth.repository";
+import validation from "../../Utils/ValidationUtil/validation"
+import {
+    BadRequestError, 
+    NotFoundError, 
+    UnAuthenticatedError
+} from "../../Utils/ErrorUtil"
+import bcrypt from "../../Utils/bcrypt.encrypt"
+
+
+// instances
+const authRepository = new AuthRepository;
+const validations = new validation;
+
+
+export default class AuthService {
+    public async registerUser(req: Request, res: Response, next: NextFunction) {
```

Comment:
This function is not modular...It's doing alot at the same time. You can breake some functionality of it into some other sub functions